app/src/main/java/com/google/maps/android/compose/MapInColumnActivity.kt

```diff
@@ -0,0 +1,224 @@
+        super.onCreate(savedInstanceState)
+        setContent {
+            // Observing and controlling the camera's state can be done with a CameraPositionState
+            val cameraPositionState = rememberCameraPositionState {
+                position = defaultCameraPosition
+            }
+            var columnScrollingEnabled by remember { mutableStateOf(true) }
+
+            // Use a LaunchedEffect keyed on the camera moving state to enable column scrolling when the camera stops moving
+            LaunchedEffect(cameraPositionState.isMoving) {
+                if (!cameraPositionState.isMoving) {
+                    columnScrollingEnabled = true
+                    Log.d(TAG, "Map camera stopped moving - Enabling column scrolling...")
+                }
+            }
+
+            MapInColumn(
+                modifier = Modifier.fillMaxSize(),
+                cameraPositionState,
+                columnScrollingEnabled = columnScrollingEnabled,
+                onMapTouched = {
+                    columnScrollingEnabled = false
+                    Log.d(
+                        TAG,
+                        "User touched map - Disabling column scrolling after user touched this Box..."
+                    )
+                },
+                onMapLoaded = { }
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun MapInColumn(
+    modifier: Modifier = Modifier,
+    cameraPositionState: CameraPositionState,
+    columnScrollingEnabled: Boolean,
+    onMapTouched: () -> Unit,
+    onMapLoaded: () -> Unit,
+) {
+    Surface(
+        modifier = modifier,
+        color = MaterialTheme.colors.background
+    ) {
+        var isMapLoaded by remember { mutableStateOf(false) }
+
+        Column(
+            Modifier
+                .fillMaxSize()
+                .verticalScroll(
+                    rememberScrollState(),
+                    columnScrollingEnabled
+                ),
+            horizontalAlignment = Alignment.Start
+        ) {
+            Spacer(modifier = Modifier.padding(10.dp))
+            for (i in 1..20) {
+                Text(
+                    text = "Item $i",
+                    modifier = Modifier
+                        .padding(start = 10.dp)
+                        .testTag("Item $i")
+                )
+            }
+            Spacer(modifier = Modifier.padding(10.dp))
+
+            Box(
+                Modifier
+                    .fillMaxWidth()
+                    .height(200.dp)
+                    .testTag("Map")
+            ) {
+                GoogleMapViewInColumn(
+                    modifier = Modifier
+                        .fillMaxSize(),
+                    cameraPositionState = cameraPositionState,
+                    onMapLoaded = {
+                        isMapLoaded = true
+                        onMapLoaded()
+                    },
+                )
+                Box(
```

Comment:
I'm wondering if we can do without the invisible `Box` layout—curious if you tried this technique on the parent `Box` layout or on the `GoogleMap` composable directly?

Reply:
Ah, thanks for pointing this out! I went through a lot of iterations and thought I tried this, but apparently not, because it does work without the box. I updated it in https://github.com/googlemaps/android-maps-compose/pull/94/commits/c2268fe8fb81d1267c1c1cc19eb158925ad94791 to use `.pointerInteropFilter()` directly on `GoogleMapViewInColumn()`:

```kotlin
        GoogleMapViewInColumn(
		modifier = Modifier
			.fillMaxSize()
			.testTag("Map")
			.pointerInteropFilter(
				onTouchEvent = {
				...
```

I was hoping this would fix the test FIXME issue too, but unfortunately not...